apis/python/src/tiledbsc/soma.py

```diff
@@ -315,9 +329,18 @@ def query(
         # obs, but no rows with cell_type == "blood".
         if slice_obs_df is None:
             return None
-        obs_ids = list(slice_obs_df.index)
-        if len(obs_ids) == 0:
+        if len(slice_obs_df.index) == 0:
             return None
+        # At the tiledb multi-index level, if we're say slicing on obs_ids but not var_ids,
+        # we'll do `A.df[obs_ids, :]`. We can't pass a `:` down the callstack to get there,
+        # but we pass `None` instead.
+        #
+        # It's important to do this. Say for example the X matrix is nobs=1000 by nvar=2000,
```

Comment:
We were doing the following:

* For example a 1000x2000 matrix
* 80 obs IDs -> 80 obs IDs
* `None` (for `:`) -> 2000 var IDs

is not new. However, it performed OK with `allow_duplicates=True` but performed poorly with `allow_duplicates=False`.